x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/RollupActionIT.java

```diff
@@ -7,82 +7,236 @@
+        );
+        Map<String, Phase> phases = Map.of("hot", new Phase("hot", TimeValue.ZERO, hotActions));
+        LifecyclePolicy lifecyclePolicy = new LifecyclePolicy(policy, phases);
+        Request createPolicyRequest = new Request("PUT", "_ilm/policy/" + policy);
+        createPolicyRequest.setJsonEntity("{ \"policy\":" + Strings.toString(lifecyclePolicy) + "}");
+        client().performRequest(createPolicyRequest);
+
+        // and a template
+        Request createTemplateRequest = new Request("PUT", "_template/" + index);
+        createTemplateRequest.setJsonEntity("""
+            {
+              "index_patterns": ["%s-*"],
+              "settings": {
+                "number_of_shards": %s,
+                "number_of_replicas": 0,
+                "index.lifecycle.name": "%s",
+                "index.lifecycle.rollover_alias": "%s"
+              }
+            }""".formatted(index, 1, policy, alias));
+        createTemplateRequest.setOptions(expectWarnings(RestPutIndexTemplateAction.DEPRECATION_WARNING));
+        client().performRequest(createTemplateRequest);
+
+        // then create the index and index a document to trigger rollover
+        createIndex(originalIndex, alias);
+        index(
             client(),
-            index,
-            alias,
-            Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1).put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            originalIndex,
+            true,
+            null,
+            "@timestamp",
+            "2020-01-01T05:10:00Z",
+            "volume",
+            11.0,
+            "metricset",
+            randomAlphaOfLength(5)
         );
-        index(client(), index, "_id", "timestamp", "2020-01-01T05:10:00Z", "volume", 11.0);
-        RollupActionConfig rollupConfig = new RollupActionConfig(DateHistogramInterval.DAY);
 
-        createNewSingletonPolicy(client(), policy, "cold", new RollupILMAction(rollupConfig, policy));
-        updatePolicy(client(), index, policy);
+        assertBusy(() -> assertNotNull("Cannot retrieve rollup index name", getRollupIndexName(originalIndex)), 30, TimeUnit.SECONDS);
+        String rollupIndex = getRollupIndexName(originalIndex);
 
-        assertBusy(() -> assertNotNull(getRollupIndexName(index)));
-        String rollupIndex = getRollupIndexName(index);
-        assertBusy(() -> assertTrue(indexExists(rollupIndex)));
-        assertBusy(() -> assertThat(getOnlyIndexSettings(client(), rollupIndex).get(LifecycleSettings.LIFECYCLE_NAME), equalTo(policy)));
-        assertBusy(() -> assertTrue(indexExists(index)));
+        assertBusy(() -> assertTrue("Rollup index does not exist", indexExists(rollupIndex)), 30, TimeUnit.SECONDS);
+        assertBusy(() -> assertFalse("Source index should have been deleted", indexExists(originalIndex)), 30, TimeUnit.SECONDS);
+        // assertBusy(() -> assertThat(getStepKeyForIndex(client(), rollupIndex), equalTo(PhaseCompleteStep.finalStep("hot").getKey())));
```

Comment:
Not sure if we should test this too. Right now RollupILMAction does not set ILM phase in the index. should this be done?